src/ByteCountingStream.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace GuzzleHttp\Psr7;
+
+use Psr\Http\Message\StreamInterface;
+use SebastianBergmann\GlobalState\RuntimeException;
+
+/**
+ *  Stream decorator that counts the number of bytes read off of an underlying
+ *  read stream.
```

Comment:
Can you add more information here? Let's call out the fact that this stream decorator is used to ensure that an expected number of bytes can read from an underlying stream. If the underlying stream does not provide the expected number of bytes, then a `\RuntimeException` is thrown. The decorator will only up to the expected number of bytes. Any additional data on the stream will be ignored.